thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiDialogFragment.java

```diff
@@ -74,144 +65,83 @@ public P getPresenter() {
-        }
-
-        final TiConfiguration config = mPresenter.getConfig();
-        if (!destroyPresenter &&
-                !config.shouldRetainPresenter()) {
-            // configuration says the presenter should not be retained, a new presenter instance
-            // will be created and the current presenter should be destroyed
-            destroyPresenter = true;
-            TiLog.v(TAG, "presenter configured as not retaining, destroying " + mPresenter);
-        }
-
-        if (!destroyPresenter &&
-                !config.useStaticSaviorToRetain()
-                && AndroidDeveloperOptions.isDontKeepActivitiesEnabled(getActivity())) {
-            // configuration says the PresenterSavior should not be used. Retaining the presenter
-            // relays on the Activity nonConfigurationInstance which is always null when
-            // "don't keep activities" is enabled.
-            // a new presenter instance will be created and the current presenter should be destroyed
-            destroyPresenter = true;
-            TiLog.v(TAG, "the PresenterSavior is disabled and \"don\'t keep activities\" is "
-                    + "activated. The presenter can't be retained. Destroying " + mPresenter);
-        }
-
-        if (destroyPresenter) {
-            mPresenter.destroy();
-            PresenterSavior.INSTANCE.free(mPresenterId);
-        } else {
-            TiLog.v(TAG, "not destroying " + mPresenter
-                    + " which will be reused by the next Activity instance, recreating...");
-        }
+        mDelegate.onDestroy_afterSuper();
     }
 
     @Override
     public void onDestroyView() {
-        mPresenter.detachView();
+        mDelegate.onDestroyView_beforeSuper();
         super.onDestroyView();
     }
 
     @Override
     public void onSaveInstanceState(final Bundle outState) {
         super.onSaveInstanceState(outState);
-        outState.putString(SAVED_STATE_PRESENTER_ID, mPresenterId);
+        mDelegate.onSaveInstanceState_afterSuper(outState);
     }
 
     @Override
     public void onStart() {
         super.onStart();
-        mActivityStarted = true;
-
-        if (isUiPossible()) {
-            getActivity().getWindow().getDecorView().post(new Runnable() {
-                @Override
-                public void run() {
-                    if (isUiPossible() && mActivityStarted) {
-                        mViewBinder.bindView(mPresenter, TiDialogFragment.this);
-                    }
-                }
-            });
-        }
+        mDelegate.onStart_afterSuper();
     }
 
     @Override
     public void onStop() {
-        mActivityStarted = false;
-        mPresenter.detachView();
+        mDelegate.onStop_beforeSuper();
         super.onStop();
     }
 
+    @Override
+    public boolean postToMessageQueue(final Runnable runnable) {
+        return getActivity().getWindow().getDecorView().post(runnable);
```

Comment:
true. It works for us because it's only called in `onStart()`. But sadly this is public API and should always work. I suggest creating a separate PR for this